cookbook/validation/validation-group-service-resolver.rst

```diff
@@ -0,0 +1,55 @@
+How to dynamically configure validation groups
+=============================================
+
+If you need some advanced logic to determine the validation groups that cannot
+be simply determined by a simple callback (for example that requires a few 
+services) you can use a service. Implement a service that implements ``__invoke``
+which accepts a FormInterface as a parameter.
+
+.. code-block:: php
+
+    use Symfony\Component\Form\FormInterface;
+    
+    class ValidationGroupResolver
+    {
+        private $service1;
+        private $sevice2;
+        public function __construct($service1,$service2)
+        {
+            $this->service1 = $service1;
+            $this->service2 = $service2;
+        }
+        
+        public function __invoke(FormInterface $form)
+        {
+            //figure out what groups to apply and return an array
+        }
+    }
+
+Then in your form, inject the resolver and set it as the ``validation_groups``.
+
+.. code-block:: php
+
+    use Symfony\Component\OptionsResolver\OptionsResolver;
+    use Symfony\Component\Form\AbstractType
+    
+    class MyClassType extends AbstractType
+    {
+        private $groupResolver;
+        public function __construct($groupResolver)
+        {
+            $this->groupResolver = $groupResolver;
+        }
+        
+        // ...
+        public function configureOptions(OptionsResolver $resolver)
```

Comment:
Isn't this a 2.3 feature? If so, we should use the 2.3 API here (`setDefaultOptions` and `OptionsResolverInterface`). You don't have to create a new PR though